hazelcast-sql/src/test/java/com/hazelcast/sql/misc/SqlNoSerializationTest.java

```diff
@@ -54,66 +66,61 @@
 @RunWith(Parameterized.class)
 @Parameterized.UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
-public class SqlNoSerializationTest extends SqlTestSupport {
+public class SqlNoSerializationTest extends OptimizerTestSupport {
 
     private static final String MAP_NAME = "map";
     private static final int KEY_COUNT = 100;
 
     @Parameterized.Parameter
     public boolean useIndex;
 
-    private final TestHazelcastFactory factory = new TestHazelcastFactory();
-    private HazelcastInstance member;
-
     private static volatile boolean failOnSerialization;
 
     @Parameterized.Parameters(name = "useIndex:{0}")
     public static Collection<Object> parameters() {
-        return Arrays.asList(true, false);
+        return Arrays.asList(false, true);
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        initialize(1, smallInstanceConfig());
     }
 
     @Before
     public void before() {
-        member = factory.newHazelcastInstance();
-        factory.newHazelcastInstance();
-
         Map<Key, Value> localMap = new HashMap<>();
 
         for (int i = 0; i < KEY_COUNT; i++) {
             localMap.put(new Key(i), new Value(i));
         }
 
-        IMap<Key, Value> map = member.getMap(MAP_NAME);
+        IMap<Key, Value> map = instance().getMap(MAP_NAME);
 
         map.putAll(localMap);
 
         // An index may change the behavior due to MapContainer.isUseCachedDeserializedValuesEnabled.
         if (useIndex) {
-            map.addIndex(new IndexConfig().setType(IndexType.SORTED).addAttribute("val"));
```

Comment:
Is there a reason `SORTED` has been replaced with `HASH`?

Reply:
Sure, because `test()` makes point lookup, which is perfect `HASH` index use case.